x-pack/plugins/licensing/public/plugin.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  CoreSetup,
+  CoreStart,
+  Plugin as CorePlugin,
+  PluginInitializerContext,
+} from 'src/core/public';
+import { Poller } from '../../../../src/core/utils/poller';
+import { LicensingPluginSetup, ILicensingPlugin } from '../common/types';
+import {
+  API_ROUTE,
+  LICENSING_SESSION,
+  LICENSING_SESSION_SIGNATURE,
+  SIGNATURE_HEADER,
+  DEFAULT_POLLING_FREQUENCY,
+} from '../common/constants';
+import { License } from '../common/license';
+import { hasLicenseInfoChanged } from '../common/has_license_info_changed';
+
+export class Plugin implements CorePlugin<LicensingPluginSetup>, ILicensingPlugin {
+  private core!: CoreSetup;
+  private poller!: Poller<License>;
+  private refresher?: Promise<void>;
+  private removeInterceptor!: () => void;
+  private signature = '';
+  public pollingFrequency = DEFAULT_POLLING_FREQUENCY;
+
+  constructor(context: PluginInitializerContext) {}
+
+  private getSession() {
+    const raw = sessionStorage.getItem(LICENSING_SESSION);
+    const signature = sessionStorage.getItem(LICENSING_SESSION_SIGNATURE);
+    const json = raw ? JSON.parse(raw) : {};
+
+    if (signature) {
+      this.signature = signature;
+    }
+
+    return json;
+  }
+
+  private setSession(license: any, signature?: string) {
+    sessionStorage.setItem(LICENSING_SESSION, JSON.stringify(license));
+
+    if (signature) {
+      sessionStorage.setItem(LICENSING_SESSION_SIGNATURE, signature);
+    }
+  }
+
+  private clearSession() {
+    sessionStorage.removeItem(LICENSING_SESSION_SIGNATURE);
+    sessionStorage.removeItem(LICENSING_SESSION);
```

Comment:
no operations with sessionStorage are tested

Reply:
Could you elaborate on what you mean by this?

Reply:
there are no tests that data are stored between page reloads.